test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -156,17 +156,29 @@ def _test_common(
     ):
         counters.clear()
         torch._dynamo.reset()
-        if (
-            check_autocast == torch.bfloat16
-            and torch.ops.mkldnn._is_mkldnn_bf16_supported()
+        has_xpu = any(
+            isinstance(input, torch.Tensor) and input.device.type == "xpu"
+            for input in inputs
+        )
+        if check_autocast == torch.bfloat16 and (
+            torch.ops.mkldnn._is_mkldnn_bf16_supported() or has_xpu
```

Comment:
```suggestion
        has_xpu = any(
            isinstance(input, torch.Tensor) and input.device.type == "xpu"
            for input in inputs
        )
        device_type = 'xpu' if has_xpu else 'cpu'
        if torch.ops.mkldnn._is_mkldnn_bf16_supported() or torch.ops.mkldmm._is_mkldnn_fp16_supported():
            maybe_autocast = torch.amp.autocast(device_type, check_autocast)
```
`torch.cpu.amp.autocast` is deprecated now. Use `torch.amp.autocast` instead.

Reply:
modified